flake.nix

```diff
@@ -0,0 +1,140 @@
+
+                        # Web
+                        export WEB_BASEURL="http://localhost:5173"
+
+                        # Minio
+                        export MINIO_ENDPOINT="localhost:9000"
+                        export MINIO_DATA_DIR="$STATEDIR/minio/data"
+                        export MINIO_CONFIG_DIR="$STATEDIR/minio/config"
+                        export MINIO_ROOT_USER="minioadmin"
+                        export MINIO_ROOT_PASSWORD="minioadmin"
+                        export MINIO_USE_SSL="false"
+                        export MC_CONFIG_DIR="$STATEDIR/minio/mc"
+
+                        mkdir -p "$MC_CONFIG_DIR"
+                        cat <<EOF > "$MC_CONFIG_DIR/config.json"
+                        {
+                            "version": "10",
+                            "aliases": {
+                                "local": {
+                                    "url": "http://$MINIO_ENDPOINT",
+                                    "accessKey": "$MINIO_ROOT_USER",
+                                    "secretKey": "$MINIO_ROOT_PASSWORD",
+                                    "api": "S3v4",
+                                    "path": "auto"
+                                }
+                            }
+                        }
+                        EOF
+
+                        #API
+                        export VAULT_ORG_ID="25ee0d1b-4e14-42f8-bf34-d3191ff704b7"
+                        export VAULT_PROJECT_ID="3196ec06-6d59-4fc4-affa-fa3fdd12cd1c"
+                        export VAULT_APP_NAME="salz"
+                        hcp profile set organization_id "$VAULT_ORG_ID" --quiet
+                        hcp profile set project_id "$VAULT_PROJECT_ID" --quiet
+                        hcp profile set vault-secrets/app "$VAULT_APP_NAME" --quiet
+                        hcp auth print-access-token >/dev/null 2>&1 || { echo >&2 "Not logged into HCP -- login via hcp auth login"; exit 1; }
+                        export OAUTH_CLIENT_KEY=$(hcp vault-secrets secrets open github_oauth_client_id --format json | jq -r '.static_version.value')
+                        export OAUTH_CLIENT_SECRET=$(hcp vault-secrets secrets open github_oauth_client_secret --format json | jq -r '.static_version.value')
+                        export PG_URI="postgres://$PGUSER:$PGPASSWORD@localhost:5432/$PGDATABASE?sslmode=disable"
+                        export MINIO_ACCESS_KEY="$MINIO_ROOT_USER"
+                        export MINIO_SECRET_KEY="$MINIO_ROOT_PASSWORD"
+
+
+                        # process Compose
+                        export PC_PORT_NUM=8081
+
+                        trap \
+                            "
+                            cd $PWD
+                            rm -rf $NIX_SHELL_DIR
+                            process-compose down
+                            " \
+                            EXIT
+
+                        process-compose -D -f $PWD/process-compose.yaml
```

Comment:
I don't actually like having process compose start when running `nix develop`. How strongly do you feel about leaving it in?

Reply:
I would rather call `process-compose` after entering the development shell. Idk. Sometimes I just want to poke around and don't necessarily want to spin up the entire stack.

Reply:
I don't feel strongly about this at all, I'd be happy to remove it.